source/leuven/services/openondemand.rst

```diff
@@ -191,136 +226,244 @@ of using any of the other partitions. There are however some limitations on the
+If you indeed need multiple nodes or full GPUs to test your code/program, go ahead and request the resources for
+your interactive app from a more suited partition.
+In the case that you have passed the testing phase, and you want to start conducting experiments;
+we recommend that you make the switch to batch jobs instead, as they will not require your presence to start your code.
 
 .. _interactive_shell:
 
 Interactive shell
 -----------------
 
-This app will launch a shell on (one of) the requested node(s), allowing you to use these compute resources from within a Linux terminal. This is different
-than the shell you get in the "Clusters" menu, which directs you towards one of the login nodes.
+This app will launch a shell on (one of) the requested node(s), allowing you to use these compute resources
+from within a Linux terminal.
+This is different than the shell you get in the "Clusters - Login Server Shell Access" menu,
+which directs you towards one of the login nodes.
 
-Jupyter Lab
+JupyterLab
 -----------
 
-With this app you can use and create Jupyter Notebooks. This can be handy both for R and Python coding. There are two kernels already available, being a Python and a
-R kernel. The Python and R versions that are used for this, are the versions located in ``/usr/bin``. While you can use these to do some testing, it is not recommended 
-to work with these. It is better to work with conda environments in this case. You should :ref:`install miniconda <install_miniconda_python>` if you have not installed it yet.
-When you do not have any conda environments and their associated kernels, both the Python and R installation will default to the ``~/miniconda3/bin/...`` installation.
+With this app you can create or (re)run Jupyter Notebooks. This can be handy both for R and Python coding.
+In the app resource form, besides the normal choices (:ref:`listed above <interactive-apps>`), you can also choose
+a 'Toolchain year' such as '2023a' from a drop-down menu.
+Based on that choice, a correct JupyterLab module will be loaded together with its dependencies (specifically Python).
+To reproduce your numerical results at any time in the future, make sure you stick to choosing the same 'Toolchain year'.
+
+Furthermore, you may optionally choose to load ``SciPy-bundle`` (for widely-used packages like ``scipy``,
+``numpy``, ``pandas`` and more) and/or ``matplotlib`` in your environment from the same 'Toolchain year'.
+
+Once you launch a JupyterLab session, a default kernel called ``Python 3 (ipykernel)`` is already available in your session.
+This kernel, in addition to the Python standard library, would enable using extra packages from
+``SciPy-bundle`` and/or ``matplotlib``, if you already selected them.
+
+If the "standard" environment explained above does not provide all packages that you need, then
+it is recommended to manage your custom-made R or Python environments using Conda.
```

Comment:
```suggestion
it is recommended to manage your custom-made R or Python environments using Conda or Python virtual environments.
```

Reply:
Sure, but I did it slightly differently now.